src/org/exist/protocolhandler/embedded/EmbeddedDownloadThread.java

```diff
@@ -38,29 +39,32 @@
  */
 public class EmbeddedDownloadThread extends Thread {
     
-    private final static Logger logger = LogManager.getLogger(EmbeddedDownloadThread.class);
+    private static final Logger LOG = LogManager.getLogger(EmbeddedDownloadThread.class);
     
-    private XmldbURL xmldbURL;
-    private OutputStream bos;
+    private final XmldbURL xmldbURL;
+    private final OutputStream bos;
     
-    private Subject subject = null;
+    private Subject subject;
     private BrokerPool brokerPool;
+
+    private static final AtomicInteger threadInitNumber = new AtomicInteger();
     
     /**
      *  Constructor of EmbeddedDownloadThread.
      * 
      * @param url Document location in database.
      * @param bos Stream to which the document is written.
      */
-    public EmbeddedDownloadThread(XmldbURL url, OutputStream bos) {
-        xmldbURL = url;
+    public EmbeddedDownloadThread(final XmldbURL url, final OutputStream bos) {
+        super("EmbeddedDownloadThread-" + threadInitNumber.getAndIncrement());
+        this.xmldbURL = url;
         this.bos = bos;
         
         try {
-            BrokerPool pool = BrokerPool.getInstance(url.getInstanceName());
-            subject = pool.getActiveBroker().getCurrentSubject();
-        } catch (Throwable e) {
-            //e.printStackTrace();
+            final BrokerPool pool = BrokerPool.getInstance(url.getInstanceName());
+            this.subject = pool.getActiveBroker().getCurrentSubject();
```

Comment:
I see that you do this on more locations..... not needed, but is it a good practice I don;t know of?